azurelinuxagent/common/protocol/wire.py

```diff
@@ -661,7 +662,13 @@ def fetch_manifest(self, version_uris):
         version_uris_shuffled = version_uris
         random.shuffle(version_uris_shuffled)
 
+        start_time = datetime.now()
         for version in version_uris_shuffled:
+            
+            if datetime.now() - start_time > timedelta(minutes=5):
```

Comment:
Can you add a comment here suggesting why we're doing this

Reply:
You mean `timeout=5` for fetch manifest itself, right?

Reply:
Actually yeah parameterizing it might even be better. I just meant adding a global variable but your idea is better haha!